app/src/main/java/com/vero/tasky/agenda/domain/worker/SyncAgendaWorker.kt

```diff
@@ -30,10 +32,16 @@ class SyncAgendaWorker @AssistedInject constructor(
         val result = agendaRepository.syncAgenda(
             deletedEventIds = agendaItems.filter { it.type == AgendaItemType.Event }.map { it.id },
             deletedTaskIds = agendaItems.filter { it.type == AgendaItemType.Task }.map { it.id },
-            deletedReminderIds =agendaItems.filter { it.type == AgendaItemType.Reminder }.map { it.id },
+            deletedReminderIds = agendaItems.filter { it.type == AgendaItemType.Reminder }.map { it.id },
         )
-        return if (result.isSuccess)
+        return if (result.isSuccess) {
+            supervisorScope {
+                agendaItems.map {
+                    launch { dao.deleteAgendaItem(it) }
+                }.forEach { it.join() }
```

Comment:
I'm pretty sure the ".forEach { it.join() }" can be replaced with a ".joinAll()"